contracts/SortedVessels.sol

```diff
@@ -360,23 +354,20 @@ contract SortedVessels is OwnableUpgradeable, ISortedVessels {
 		uint256 _NICR,
 		address _startId
 	) internal view returns (address, address) {
+		Data storage assetData = data[_asset];
+
 		// If `_startId` is the head, check if the insert position is before the head
-		if (
-			data[_asset].head == _startId && _NICR >= _vesselManager.getNominalICR(_asset, _startId)
-		) {
+		if (assetData.head == _startId && _NICR >= _vesselManager.getNominalICR(_asset, _startId)) {
 			return (address(0), _startId);
 		}
 
 		address prevId = _startId;
-		address nextId = data[_asset].nodes[prevId].nextId;
+		address nextId = assetData.nodes[prevId].nextId;
 
 		// Descend the list until we reach the end or until we find a valid insert position
-		while (
-			prevId != address(0) &&
-			!_validInsertPosition(_asset, _vesselManager, _NICR, prevId, nextId)
-		) {
-			prevId = data[_asset].nodes[prevId].nextId;
-			nextId = data[_asset].nodes[prevId].nextId;
+		while (prevId != address(0) && !_validInsertPosition(_asset, _vesselManager, _NICR, prevId, nextId)) {
+			prevId = assetData.nodes[prevId].nextId;
+			nextId = assetData.nodes[prevId].nextId;
```

Comment:
Can be improved (assetData.nodes[prevId] repeated)

Reply:
Nope, look closely and you'll notice that prevId changes between those two lines, so we cannot cache the lookup.